internal/juju/offers.go

```diff
@@ -192,6 +193,10 @@ func (c offersClient) DestroyOffer(input *DestroyOfferInput) error {
 
 	client := applicationoffers.NewClient(conn)
 	offer, err := client.ApplicationOffer(input.OfferURL)
+	if jujuerrors.Is(err, jujuerrors.NotFound) {
```

Comment:
We don't get actual types out of the juju clients usually. Check out some of the other NotFound code.